test/e2e/catalog_e2e_test.go

```diff
@@ -31,7 +31,7 @@ import (
 var _ = Describe("Catalog", func() {
 	It("loading between restarts", func() {
 
-		defer cleaner.NotifyTestComplete(true)
+		defer TearDown(testNamespace)
```

Comment:
> Can these all turn into one AfterEach since you're touching them anyway?

Yeah, can do.

>  but if this cleanup behavior is supposed to be idempotent, that seems like an overcomplication.

I agree, but I think we need to consider cleaning these up carefully though. Not all tests use `testNamespace`, but instead generate one or more namespaces on the fly, which is what I think some of these `defer` statements service. I n general, I think we should standardize on one of the two above patterns across all of our tests. Suggestions?

Reply:
Can we use the TestContext to provide namespaces to tests? We'd be able to make cleanup, randomization, etc. transparent to individual tests.